tune_sklearn/_trainable.py

```diff
@@ -314,3 +320,100 @@ def reset_config(self, new_config):
         self.config = new_config
         self._setup(new_config)
         return True
+
+
+class _PipelineTrainable(_Trainable):
+    """Class to be passed in as the first argument of tune.run to train models.
+
+    Overrides Ray Tune's Trainable class to specify the setup, train, save,
+    and restore routines.
+
+    Runs all checks and configures parameters for the last step of the Pipeline.
+
+    """
+
+    @property
+    def base_estimator_name(self):
+        return self.main_estimator.steps[-1][0]
+
+    @property
+    def base_estimator(self):
+        return self.main_estimator.steps[-1][1]
+
+    def _can_partial_fit(self):
+        return check_partial_fit(self.base_estimator)
+
+    def _can_warm_start_iter(self):
+        return check_warm_start_iter(self.base_estimator)
+
+    def _can_warm_start_ensemble(self):
+        return check_warm_start_ensemble(self.base_estimator)
+
+    def _is_xgb(self):
+        return is_xgboost_model(self.base_estimator)
```

Comment:
since the property is overriden you'll be able to remove these right?

Reply:
It has a different name now. I didn't want to override it completely, because there are places where the entire Pipeline needs to be used.

Reply:
In fact, the entire Pipeline should be used by default, and the last step should be only used in specific places

Reply:
Got it!